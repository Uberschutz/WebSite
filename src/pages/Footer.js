import React, { Component } from 'react';
import '../styles/bootstrap.css';
import '../styles/HomePage.css';

class Footer extends Component {
    render() {
        return (
            <div className="uber-color form-align">
                <button type="button" className="btn btn-light button-footer">Contact</button>
                <br/>
                <span>
                    Â© 2019
                </span>
                <br/>
                <span>
                    Created by Marianne LEVEE
                </span>
            </div>
        )
    }
}

export default Footer;