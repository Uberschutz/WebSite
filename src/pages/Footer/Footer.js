import React, { Component } from 'react';
import '../../styles/bootstrap.css';
import '../../styles/HomePage.css';

import { Link } from 'react-router-dom';
import {displayContent} from "../../utils/translationDisplay";

class Footer extends Component {
	constructor(props) {
		super(props);

		this.state = {
			lang: 'fr'
		}
	}

	componentDidMount() {
		if (this.props.base) {
			const {base: {language}} = this.props;
			console.log(language, this.state.lang, 'kek')
			if (this.state.lang !== language) {
				this.setState({
					lang: language
				})
			}
		}
	}

	componentDidUpdate(prevProps, prevState, snapshot) {
		// console.log(prevProps, this.props);
		this.props.base.language !== prevProps.base.language && this.setState({lang: this.props.base.language}, () => console.log('re'));
	}

    render() {
		let i = 0;
        return (
            <div className="uber-color2 form-align footer">
                <Link to ='/Contact+FAQ'>
                    <button type="button" className="btn btn-primary button-footer">Contact</button>
                </Link>
                <br/>
                <span>
                    Â© 2018 - 2020
                </span>
                <br/>
                <span>
                    {displayContent(this.state.lang, i++, 'footer')}<br/>{displayContent(this.state.lang, i++, 'footer')}<a href="https://icones8.fr">icons8</a> <br/>
                </span>
            </div>
        )
    }
}

export default Footer;