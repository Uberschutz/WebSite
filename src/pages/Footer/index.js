import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import * as actions from '../../actions'
import Footer from './Footer'

const mapStateToProps = state => ({
	base: state.base,
})

const mapDispatchToProps = dispatch => ({
	setLanguage: bindActionCreators(actions.setLanguage, dispatch)
})

const FooterContainer = connect(mapStateToProps, mapDispatchToProps)(Footer)

export default FooterContainer