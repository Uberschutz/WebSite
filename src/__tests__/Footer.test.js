import React from 'react';
import ReactDOM from 'react-dom';
import Footer from '../pages/Footer/Footer';
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";

import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });

describe('Testing the Footer component rendering', () => {

    it('renders Footer without crashing', () => {
        const div = document.createElement('div');
        // const paramsComponent = shallow(<Router><Footer/></Router>);
        ReactDOM.render(<Router><Footer/></Router>, div);
        expect(div).toMatchSnapshot();
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders Footer in english', () => {
        const div = document.createElement('div');
        const props = {
            base: {
                language: 'en'
            }
        };
        // const paramsComponent = shallow(<Router><Footer {...props}/></Router>);
        ReactDOM.render(<Router><Footer {...props}/></Router>, div);
        expect(div).toMatchSnapshot();
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders Footer in french', () => {
        const div = document.createElement('div');
        const props = {
            base: {
                language: 'fr'
            }
        };
        // const paramsComponent = shallow(<Router><Footer {...props}/></Router>);
        ReactDOM.render(<Router><Footer {...props}/></Router>, div);
        expect(div).toMatchSnapshot();
        ReactDOM.unmountComponentAtNode(div);
    });
});
