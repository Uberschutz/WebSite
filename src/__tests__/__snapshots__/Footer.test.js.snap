// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Footer without crashing 1`] = `
<div>
  <div
    class="uber-color2 form-align"
  >
    <a
      href="/Contact+FAQ"
    >
      <button
        class="btn btn-primary button-footer"
        type="button"
      >
        Contact
      </button>
    </a>
    <br />
    <span>
      Â© 2019
    </span>
    <br />
    <span>
      Created by Marianne LEVEE & Thomas DEROTE-PARCELLIER
    </span>
  </div>
</div>
`;
